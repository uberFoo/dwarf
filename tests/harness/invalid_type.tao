struct Foo {
    a: i32,
    c: int,
    b: Bar,
}

impl Foo {
    // In order to test the bad parameters we need to have a valid return typ.
    fn foo(self, a: f64, b: bool, c: u8) {}

    // Here we can test the borked return type.
    fn bar(self) -> Bar {}

    // And here we can test something wrong with the return value.
    fn baz() -> Foo {
        Baz {}
    }
}

fn foo(a: f64, b: bool, c: u8) {}

// Here we can test the borked return type.
fn bar() -> Bar {}

// And here we can test something wrong with the return value.
fn baz() -> Foo {
    Baz {}
}

// Same sort of thing as above.

fn main() {
    let a: i32 = 1;
}
