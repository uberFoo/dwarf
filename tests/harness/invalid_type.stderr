[31mError:[0m Unknown type: `i32`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0minvalid_type:2:8[38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m2 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ma[0m[38;5;249m:[0m[38;5;249m [0m[31mi[0m[31m3[0m[31m2[0m[31m,[0m
 [38;5;240m  â”‚[0m        [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m  â”‚[0m          [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m unknown type [36mi32[0m
 [38;5;240m  â”‚[0m 
 [38;5;240m  â”‚[0m [38;5;115mNote[0m: try `[33mint[0m`
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m Unknown type: `Bar`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0minvalid_type:4:8[38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m4 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249mb[0m[38;5;249m:[0m[38;5;249m [0m[31mB[0m[31ma[0m[31mr[0m[31m,[0m
 [38;5;240m  â”‚[0m        [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m  â”‚[0m          [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m unknown type [36mBar[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m Unknown type: `f64`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0minvalid_type:9:21[38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m9 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249mf[0m[38;5;249mn[0m[38;5;249m [0m[38;5;249mf[0m[38;5;249mo[0m[38;5;249mo[0m[38;5;249m([0m[38;5;249ms[0m[38;5;249me[0m[38;5;249ml[0m[38;5;249mf[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249ma[0m[38;5;249m:[0m[38;5;249m [0m[31mf[0m[31m6[0m[31m4[0m[31m,[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249m:[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249mo[0m[38;5;249mo[0m[38;5;249ml[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249mc[0m[38;5;249m:[0m[38;5;249m [0m[38;5;249mu[0m[38;5;249m8[0m[38;5;249m)[0m[38;5;249m [0m[38;5;249m{[0m[38;5;249m}[0m
 [38;5;240m  â”‚[0m                     [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m  â”‚[0m                       [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m unknown type [36mf64[0m
 [38;5;240m  â”‚[0m 
 [38;5;240m  â”‚[0m [38;5;115mNote[0m: try `[33mfloat[0m`
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m Unknown type: `u8`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0minvalid_type:9:38[38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m9 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249mf[0m[38;5;249mn[0m[38;5;249m [0m[38;5;249mf[0m[38;5;249mo[0m[38;5;249mo[0m[38;5;249m([0m[38;5;249ms[0m[38;5;249me[0m[38;5;249ml[0m[38;5;249mf[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249ma[0m[38;5;249m:[0m[38;5;249m [0m[38;5;249mf[0m[38;5;249m6[0m[38;5;249m4[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249m:[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249mo[0m[38;5;249mo[0m[38;5;249ml[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249mc[0m[38;5;249m:[0m[38;5;249m [0m[31mu[0m[31m8[0m[31m)[0m[38;5;249m [0m[38;5;249m{[0m[38;5;249m}[0m
 [38;5;240m  â”‚[0m                                      [31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m  â”‚[0m                                       [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m unknown type [36mu8[0m
 [38;5;240m  â”‚[0m 
 [38;5;240m  â”‚[0m [38;5;115mNote[0m: try `[33mint[0m`
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m Unknown type: `Bar`.
    [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0minvalid_type:12:21[38;5;246m][0m
    [38;5;246mâ”‚[0m
 [38;5;246m12 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249mf[0m[38;5;249mn[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249ma[0m[38;5;249mr[0m[38;5;249m([0m[38;5;249ms[0m[38;5;249me[0m[38;5;249ml[0m[38;5;249mf[0m[38;5;249m)[0m[38;5;249m [0m[38;5;249m-[0m[38;5;249m>[0m[38;5;249m [0m[31mB[0m[31ma[0m[31mr[0m[31m [0m[31m{[0m[38;5;249m}[0m
 [38;5;240m   â”‚[0m                     [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m[31mâ”€[0m  
 [38;5;240m   â”‚[0m                       [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m unknown type [36mBar[0m
[38;5;246mâ”€â”€â”€â”€â•¯[0m

[31mError:[0m Unknown type: `Baz`.
    [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0minvalid_type:16:9[38;5;246m][0m
    [38;5;246mâ”‚[0m
 [38;5;246m16 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[31mB[0m[31ma[0m[31mz[0m[38;5;249m [0m[38;5;249m{[0m[38;5;249m}[0m
 [38;5;240m   â”‚[0m         [31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m   â”‚[0m          [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m unknown type [36mBaz[0m
[38;5;246mâ”€â”€â”€â”€â•¯[0m

[31mError:[0m Unknown type: `f64`.
    [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0minvalid_type:20:11[38;5;246m][0m
    [38;5;246mâ”‚[0m
 [38;5;246m20 â”‚[0m [38;5;249mf[0m[38;5;249mn[0m[38;5;249m [0m[38;5;249mf[0m[38;5;249mo[0m[38;5;249mo[0m[38;5;249m([0m[38;5;249ma[0m[38;5;249m:[0m[38;5;249m [0m[31mf[0m[31m6[0m[31m4[0m[31m,[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249m:[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249mo[0m[38;5;249mo[0m[38;5;249ml[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249mc[0m[38;5;249m:[0m[38;5;249m [0m[38;5;249mu[0m[38;5;249m8[0m[38;5;249m)[0m[38;5;249m [0m[38;5;249m{[0m[38;5;249m}[0m
 [38;5;240m   â”‚[0m           [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m   â”‚[0m             [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m unknown type [36mf64[0m
 [38;5;240m   â”‚[0m 
 [38;5;240m   â”‚[0m [38;5;115mNote[0m: try `[33mfloat[0m`
[38;5;246mâ”€â”€â”€â”€â•¯[0m

[31mError:[0m Unknown type: `u8`.
    [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0minvalid_type:20:28[38;5;246m][0m
    [38;5;246mâ”‚[0m
 [38;5;246m20 â”‚[0m [38;5;249mf[0m[38;5;249mn[0m[38;5;249m [0m[38;5;249mf[0m[38;5;249mo[0m[38;5;249mo[0m[38;5;249m([0m[38;5;249ma[0m[38;5;249m:[0m[38;5;249m [0m[38;5;249mf[0m[38;5;249m6[0m[38;5;249m4[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249m:[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249mo[0m[38;5;249mo[0m[38;5;249ml[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249mc[0m[38;5;249m:[0m[38;5;249m [0m[31mu[0m[31m8[0m[31m)[0m[38;5;249m [0m[38;5;249m{[0m[38;5;249m}[0m
 [38;5;240m   â”‚[0m                            [31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m   â”‚[0m                             [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m unknown type [36mu8[0m
 [38;5;240m   â”‚[0m 
 [38;5;240m   â”‚[0m [38;5;115mNote[0m: try `[33mint[0m`
[38;5;246mâ”€â”€â”€â”€â•¯[0m

[31mError:[0m Unknown type: `Bar`.
    [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0minvalid_type:23:13[38;5;246m][0m
    [38;5;246mâ”‚[0m
 [38;5;246m23 â”‚[0m [38;5;249mf[0m[38;5;249mn[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249ma[0m[38;5;249mr[0m[38;5;249m([0m[38;5;249m)[0m[38;5;249m [0m[38;5;249m-[0m[38;5;249m>[0m[38;5;249m [0m[31mB[0m[31ma[0m[31mr[0m[31m [0m[31m{[0m[38;5;249m}[0m
 [38;5;240m   â”‚[0m             [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m[31mâ”€[0m  
 [38;5;240m   â”‚[0m               [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m unknown type [36mBar[0m
[38;5;246mâ”€â”€â”€â”€â•¯[0m

[31mError:[0m Unknown type: `Baz`.
    [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0minvalid_type:27:5[38;5;246m][0m
    [38;5;246mâ”‚[0m
 [38;5;246m27 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[31mB[0m[31ma[0m[31mz[0m[38;5;249m [0m[38;5;249m{[0m[38;5;249m}[0m
 [38;5;240m   â”‚[0m     [31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m   â”‚[0m      [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m unknown type [36mBaz[0m
[38;5;246mâ”€â”€â”€â”€â•¯[0m

[31mError:[0m Unknown type: `i32`.
    [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0minvalid_type:33:12[38;5;246m][0m
    [38;5;246mâ”‚[0m
 [38;5;246m33 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249ma[0m[38;5;249m:[0m[38;5;249m [0m[31mi[0m[31m3[0m[31m2[0m[31m [0m[31m=[0m[38;5;249m [0m[38;5;249m1[0m[38;5;249m;[0m
 [38;5;240m   â”‚[0m            [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m[31mâ”€[0m  
 [38;5;240m   â”‚[0m              [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m unknown type [36mi32[0m
 [38;5;240m   â”‚[0m 
 [38;5;240m   â”‚[0m [38;5;115mNote[0m: try `[33mint[0m`
[38;5;246mâ”€â”€â”€â”€â•¯[0m

